composer.json

```diff
@@ -17,26 +17,29 @@
     ],
     "require": {
         "php": "^7.1",
+        "alcaeus/mongo-php-adapter": "^1.1",
         "friendsofsymfony/oauth2-php": "~1.1",
-        "symfony/framework-bundle": "~3.0|^4.0",
-        "symfony/security-bundle": "~3.0|^4.0",
-        "symfony/dependency-injection": "^2.8|~3.0|^4.0"
+        "symfony/dependency-injection": "~3.0|~4.0",
+        "symfony/framework-bundle": "~3.0|~4.0",
+        "symfony/security-bundle": "~3.0|~4.0"
     },
     "require-dev": {
         "doctrine/doctrine-bundle": "~1.0",
         "doctrine/mongodb-odm": "~1.0",
         "doctrine/orm": "~2.2",
         "phing/phing": "~2.4",
-        "php-mock/php-mock-phpunit": "^1.1",
-        "phpunit/phpunit": "~4.8|~5.0",
-        "propel/propel1": "^1.6.5",
-        "symfony/class-loader": "~3.0|^4.0",
-        "symfony/console": "~3.0|^4.0",
-        "symfony/form": "~3.0|^4.0",
-        "symfony/phpunit-bridge": "~3.0|^4.0",
-        "symfony/templating": "~3.0|^4.0",
-        "symfony/yaml": "~3.0|^4.0",
-        "willdurand/propel-typehintable-behavior": "^1.0.4"
+        "php-mock/php-mock-phpunit": "~1.0|~2.0",
+        "phpstan/phpstan-phpunit": "~0.9",
+        "phpstan/phpstan-shim": "~0.9",
+        "phpunit/phpunit": "~5.0|~6.0",
+        "propel/propel1": "~1.6",
+        "symfony/class-loader": "~3.0|~4.0",
+        "symfony/console": "~3.0|~4.0",
+        "symfony/form": "~3.0|~4.0",
+        "symfony/phpunit-bridge": "~3.0|~4.0",
+        "symfony/templating": "~3.0|~4.0",
+        "symfony/yaml": "~3.0|~4.0",
+        "willdurand/propel-typehintable-behavior": "~1.0"
```

Comment:
that change is wrong. You are changing the lowest bound of the requirement

Reply:
Thanks @stof, nice catch, will fix it.